bedwars-api/src/main/java/com/andrei1058/bedwars/api/server/VersionSupport.java

```diff
@@ -135,6 +143,73 @@ protected void loadDefaultEffects() {
      */
     public abstract boolean isInvisibilityPotion(ItemStack itemStack);
 
+    /**
+     * Check if type is a Glass type material
+     */
+    public boolean isGlass(Material type) {
+        return type != Material.AIR && (type == Material.GLASS || type.toString().contains("_GLASS"));
+    }
+
+    /**
+     * Check if block is protected by blast-proof glass or an unbreakable block from a point of view
+     * <p>
+     * if pov is null, block is checked by {@link #isGlass(Material)} and {@link IArena#isBlockPlaced(Block)} instead.
+     * Otherwise, a ray tracing is performed to check whether there is any glass block or an unbreakable block.
+     * <p>
+     * If arena is null, then glass only be checked.
+     *
+     * @param arena Arena instance.
+     * @param pov   the point of view.
+     * @param block the block instance.
+     * @param step  how frequent to check the ray (0.25 - 0.5 recommended).
+     * @return whether there's unbreakable block between the pov and the block
+     */
+    public boolean isProtected(IArena arena, Location pov, Block block, double step) {
+        if (pov == null)
+            return isGlass(block.getType()) || (arena != null && !arena.isBlockPlaced(block));
+
+        // maybe remove this?
+        if (block.getType() == Material.AIR)
```

Comment:
if all else fails the return is false, but it will save calculation time so i think it could stay.
But can a block be AIR? when called in this funciton, because if not then there is no need to check